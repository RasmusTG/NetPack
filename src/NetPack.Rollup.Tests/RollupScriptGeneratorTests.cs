using ApprovalTests;
using ApprovalTests.Reporters;
using Dazinator.AspNet.Extensions.FileProviders;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json.Linq;
using Scriban;
using System.Threading.Tasks;
using Xunit;
using Xunit.Abstractions;

namespace NetPack.Rollup.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class RollupScriptGeneratorTests
    {

        private readonly ITestOutputHelper _output;


        public RollupScriptGeneratorTests(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public async Task Can_Generate_Script()
        {

            var rollupOptions = new RollupPipeOptions();
            dynamic pluginOptions = new Newtonsoft.Json.Linq.JObject();
            pluginOptions.Start = true;
            pluginOptions.Times = 1;
            pluginOptions.Things = new JArray("foo", "bar");
            rollupOptions.AddPlugin("foopackage", pluginOptions);
            rollupOptions.AddPlugin("barpackage", null);

            var path = "./RollupTemplate.txt";
            var templateText = System.IO.File.ReadAllText(path);
            var template = Template.Parse(templateText);
            var result = template.Render(rollupOptions);

            Approvals.Verify(result);

        }     
    }
}