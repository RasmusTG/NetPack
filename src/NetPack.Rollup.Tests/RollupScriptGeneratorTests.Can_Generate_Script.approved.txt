import rollupHost from "netpack-rollup";
import RollupInputOptions from "netpack-rollup";
import RollupOutputOptions from "netpack-rollup";
import hypothetical = require("rollup-plugin-hypothetical")
import * as fs from "fs";

import plugin0 from 'foopackage';
import plugin1 from 'barpackage';